### PR TITLE
feat: setup for new icon set, v2

### PR DESCRIPTION
**Pull request by domhorvathtw**
## 🎥 🍿  
new icon set

## ❓ Context
Transferwise got a new shiny icon set. The major changes to the previous one are the following:
- `size` the new icons were designed on 16px and 24px icon grids
- `variant` each icon has an `outline` variant and some of them comes with a `filled` version as well
- `new icons` the new set includes some new icons

## 🚀 Changes
- new build scripts for generating react icon components in TS
- rollup setup with typescript to create bundles in ES6 and commonJS
- replaced icons with new icon set
- removed unused deps

Introduced a new naming convention for all icons, that encode the name, variant and the size.
e.g. `ach[outline](24)` - **< name > [variant] (size)**

Consumer apps can create an <Icon> component in their apps and use the icons like:
```tsx
// src/components/icon.tsx
import { Activity, Klarna, Linkedin, ... } from '@transferwise/icons';

export const Icon = {
  Activity, 
  Klarna, 
  Linkedin,
  ...
};

// in any other component
...
<Icon.Activity size={24} filled /> 
...
``` 

## 💬 Considerations
The new setup uses Rollup and creates two bundles for consumer apps, one with `commonJS`, and one with `ES6 modules`. The later one is **tree-shakable** if the consumer app is using at least webpack 2.

The previous setup created individual **UMD** modules for each icon component and they were imported like so:
```tsx
import TransferIcon from '@transferwise/icons/react/transfer';
```

We can add this back by tweaking the build scripts a little bit if needed.

## ✅ Checklist
- [ ] The package version is bumped according to README in `package.json`, `package-lock.json`, and `CHANGELOG.md`

## ☑️ Tasks left
- [x] Optimize SVGs with `svgo`
- [x] Test optimized SVG icons with Browserstack (IE11)
- [ ] Add option to style SVG icon with `fill` or `color` property
- [ ] Add `title` or `label` prop for accessibility reasons
- [ ] Discuss if we need to create `SVG icon sprite` with new icons in both sizes and including filled variants
- [ ] Create minified CSS file
- [ ] Remove hardcoded `fill` attributes on path elements
- [ ] add `fill="currentColor"` to each icon's path
- [ ] add demo page (WIP, tested the library with a Next.js app)
- [ ] Create pre-release `v2.0.0-beta.1`
- [ ] Update circle CI config

### CODE REVIEW

**Comment by oliverviljamaa** [comment]
Absolutely amazing job 👏

**Comment by oliverviljamaa** [comment]
> We can add this back by tweaking the build scripts a little bit if needed.
I _think_ it's not necessary as long as we support treeshaking. What's the size of the whole library?

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (oliverviljamaa) [inline .prettierrc]: Could we use TW config instead?
https://github.com/transferwise/eslint-config#4-optional-set-up-prettier-separately-to-use-our-configuration

File: icons/ach[outline](16).svg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (oliverviljamaa) [inline icons/ach[outline](16).svg]: This PR is quite tough to review due to all the changed files and GitHub not handling that many files well. Could you consider creating a separate PR adding the files first?

File: .prettierrc (hunk 1 above)
Comment (domhorvathtw) [reply .prettierrc]: yes, will add that!

File: icons/ach[outline](16).svg (hunk 2 above)
Comment (domhorvathtw) [reply icons/ach[outline](16).svg]: Yes, I will create separate PR's

**Pull request closed without merging**